gnovm/pkg/gnolang/preprocess.go

```diff
@@ -3689,6 +3703,11 @@ func tryPredefine(store Store, last BlockNode, d Decl) (un Name) {
 		})
 		d.Path = last.GetPathForName(store, d.Name)
 	case *ValueDecl:
+		if d.Type != nil {
```

Comment:
https://github.com/gnolang/gno/blob/ed919917ed1c802448487b8b4a3602db573606aa/gnovm/pkg/gnolang/preprocess.go#L501-L502

do you think these logics can be integrated to avoid excessive fragmentation?

Reply:
sine the conditions being checked are subtly different, so it seems difficult to integrate them into one. but I did modify it to use `isBlankIdentifier` function instead check manually.

https://github.com/gnolang/gno/pull/2699/commits/14543c975b9da00123f2b640655286dd75338361